### PR TITLE
 Dev/315: Spotbug

### PR DESCRIPTION
**Pull request by Johnny850807**
resolve: #315 

[WIP] Dev/315: Spotbug

### CODE REVIEW

File: Commons/src/main/java/tw/waterball/judgegirl/commons/utils/ZipUtils.java [hunk 1]
```diff
@@ -172,10 +177,14 @@ private static void writeZipEntry(String path, File file,
                                       ZipOutputStream zipos, String... ignoredFileNames) throws IOException {
         if (!contains(ignoredFileNames, file.getName())) {
             if (file.isDirectory()) {
-                for (String fileName : requireNonNull(file.list())) {
-                    writeZipEntry(path + file.getName() + "/",
-                            new File(file, fileName), zipos, ignoredFileNames);
+                String[] files = file.list();
+                if (files != null) {
+                    for (String fileName : requireNonNull(files)) {
```
Comment (Wally5077) [inline Commons/src/main/java/tw/waterball/judgegirl/commons/utils/ZipUtils.java:182]: no need 'requireNonNull'

File: Domain/Primitives/src/main/java/tw/waterball/judgegirl/primitives/problem/TestcaseIO.java [hunk 2]
```diff
@@ -83,5 +78,6 @@ public Files(String testcaseId, FileResource stdIn, FileResource stdOut,
             all.addAll(outputFiles);
             return all;
         }
+
```
Comment (Wally5077) [inline Domain/Primitives/src/main/java/tw/waterball/judgegirl/primitives/problem/TestcaseIO.java:81]: remove empty line

File: Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java [hunk 3]
```diff
@@ -214,19 +207,34 @@ protected void onBeforeRunningTestcase(Testcase testcase) {
...
-        var files = stream(requireNonNull(sandboxRootPath.toFile().listFiles()))
-                .map(File::getName).collect(Collectors.toSet());
-        files.add("std.out");
-        files.add("std.err");
-        files.add(EXECUTABLE_NAME);
-        return files;
+        File[] obj = sandboxRootPath.toFile().listFiles();
+        if (obj != null && obj.length > 0) {
+            var files = stream(obj)
+                    .map(File::getName).collect(Collectors.toSet());
```
Comment (Wally5077) [inline Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java:213]:
```Java
var files = mapToSet(obj, File::getName);
```

File: Plugins/src/main/java/tw/waterball/judgegirl/plugins/impl/cqi/CodeQualityInspectionPlugin.java [hunk 4]
```diff
@@ -103,10 +100,10 @@ private CyclomaticComplexityReport calcCyclomaticComplexity(Path sourceRootPath)
...
     }
 
-    @SneakyThrows
-    private String readSourceCode(String sourceCodePath) {
-        return Files.readString(Paths.get(sourceCodePath));
-    }
+//    @SneakyThrows
+//    private String readSourceCode(String sourceCodePath) {
+//        return Files.readString(Paths.get(sourceCodePath));
+//    }
```
Comment (Wally5077) [inline Plugins/src/main/java/tw/waterball/judgegirl/plugins/impl/cqi/CodeQualityInspectionPlugin.java:106]: why?

File: Spring-Boot/Spring-Boot-Commons/src/main/java/tw/waterball/judgegirl/springboot/advices/instrumentation/TraceIds.java [hunk 5]
```diff
@@ -25,14 +25,14 @@
     private static final int BYTES_LENGTH = 16;
     private static final int HEX_LENGTH = 2 * BYTES_LENGTH;
     private static final String INVALID = "00000000000000000000000000000000";
-    private static final Random RANDOM = new Random();
+    private static final SecureRandom random = new SecureRandom();
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Commons/src/main/java/tw/waterball/judgegirl/springboot/advices/instrumentation/TraceIds.java:28]: variable name to upper case

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: Plugins/src/main/java/tw/waterball/judgegirl/plugins/impl/cqi/CodeQualityInspectionPlugin.java (hunk 4 above)
Comment (Johnny850807) [reply Plugins/src/main/java/tw/waterball/judgegirl/plugins/impl/cqi/CodeQualityInspectionPlugin.java:106]: Keep it commented, that's not used for a while

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by timt1028 — APPROVED** [review]
LGTM

**Pull request merged**